theano/tensor/tests/test_slinalg.py

```diff
@@ -229,6 +231,27 @@ def test_solve_correctness(self):
         assert numpy.allclose(scipy.linalg.solve_triangular(U_val, b_val, lower=False),
                               upper_solve_func(U_val, b_val))
 
+    def test_solve_dtype(self):
+        if not imported_scipy:
+            raise SkipTest("Scipy needed for the Solve op.")
+
+        dtypes = ['int8', 'int16', 'int32', 'int64',
+                  'float32', 'float64']
```

Comment:
It fails with float16. Are we supposed to support it ? The output type of scipy's solve is not documented